### PR TITLE
75 promise handling

### PR DESCRIPTION
**Pull request by Frances-Hughes**
## Definition of Done

For every submission to the codebase, check that it meets the DoD below.

* [x] Code fulfils the user story and scenarios set out in the github task and UI prototype.
* [x] The feature has been thoroughly manually tested. (Ask other devs to give some fresh eyes on this as you go.)
* [x] All tests run and passed from terminal.  Any broken tests fixed, not disabled or skipped.
> `npm test`
* [x] Tests added or updated where appropriate.  (Be pragmatic about adding new tests, think; how critical is this module or function? If this function broke due to changes, how would it affect the user experience?)
* [x] eslint run and passes. Any issues fixed.
> `npm run lint`
* [x] PR created in Github.
* [x] Two peer reviews from other devs.
* [x] One peer review from one of the teachers.
* [ ] After all the peer review comments are addressed, the branch is merged to main!

### CODE REVIEW

File: client/subcomponents/volunteers/VolunteerButton/volunteerButtonHelper.js [hunk 1]
```diff
@@ -3,30 +3,29 @@ import { dispatch, getState } from '../../../store'
...
 ) {
-  const storeState = getState()
-  const { id, token } = storeState.user
-  if (!id) {
-    dispatch(showError('Please register or sign in to volunteer.'))
-  } else {
-    dispatch(setWaiting())
-    const routeMethod = willVolunteer ? 'post' : 'delete'
-    const userData = { userId: id, eventId }
+  try {
```
Comment (johnny-sainsbury) [inline client/subcomponents/volunteers/VolunteerButton/volunteerButtonHelper.js:21]: Nice work, only a few changes needed! We only wrap a try...catch block around our async calls, any code before the await needs to be outside of the try block.

File: client/subcomponents/volunteers/VolunteerButton/volunteerButtonHelper.js [hunk 2]
```diff
@@ -3,30 +3,29 @@ import { dispatch, getState } from '../../../store'
...
-        return null
-      })
-      .catch((error) => {
-        dispatch(showError(error.message))
-        return null
-      })
+      await consume('/volunteers', token, routeMethod, userData)
+      setVolunteering(willVolunteer)
+      dispatch(clearWaiting())
+      return null
```
Comment (johnny-sainsbury) [inline client/subcomponents/volunteers/VolunteerButton/volunteerButtonHelper.js]: in async...await it is no longer necessary to return null :)

File: client/subcomponents/volunteers/VolunteerButton/volunteerButtonHelper.js [hunk 3]
```diff
@@ -3,30 +3,29 @@ import { dispatch, getState } from '../../../store'
...
+    }
+  } catch (err) {
+    dispatch(showError(err.message))
+    return null
```
Comment (johnny-sainsbury) [inline client/subcomponents/volunteers/VolunteerButton/volunteerButtonHelper.js]: as above!

File: client/subcomponents/volunteers/VolunteerListItem/volunteerListItemHelper.js [hunk 4]
```diff
@@ -3,18 +3,17 @@ import { dispatch, getState } from '../../../store'
 import { setWaiting, clearWaiting } from '../../../slices/waiting'
 import { showError } from '../../../slices/error'
 
-export function toggleAttendance(volunteerData, consume = requestor) {
-  const storeState = getState()
-  const { token } = storeState.user
-  dispatch(setWaiting())
+export async function toggleAttendance(volunteerData, consume = requestor) {
+  try {
```
Comment (johnny-sainsbury) [inline client/subcomponents/volunteers/VolunteerListItem/volunteerListItemHelper.js:10]: Sweet, same as the other file:  move the try down, to above line 12

File: client/views/admin/Events/AddEvent/addEventHelper.js [hunk 5]
```diff
@@ -3,23 +3,21 @@ import { setWaiting, clearWaiting } from '../../../../slices/waiting'
...
 import requestor from '../../../../consume'
 
-export function addEvent(event, navigateTo, consume = requestor) {
-  const storeState = getState()
-  const { gardenId, token } = storeState.user
-  const newEvent = {
-    gardenId,
-    ...event,
+export async function addEvent(event, navigateTo, consume = requestor) {
+  try {
```
Comment (johnny-sainsbury) [inline client/views/admin/Events/AddEvent/addEventHelper.js:14]: same here :)

File: client/views/admin/Events/AddEvent/addEventHelper.js [hunk 6]
```diff
@@ -3,23 +3,21 @@ import { setWaiting, clearWaiting } from '../../../../slices/waiting'
...
+    const storeState = getState()
+    const { gardenId, token } = storeState.user
+    const newEvent = {
+      gardenId,
+      ...event,
+    }
+    dispatch(setWaiting())
+    await consume('/events', token, 'post', newEvent)
+    navigateTo(`/gardens/${gardenId}`)
+    return null
```
Comment (johnny-sainsbury) [inline client/views/admin/Events/AddEvent/addEventHelper.js]: same as before, remove return null

**Review by johnny-sainsbury — COMMENTED** [review]
Please make sure all of the tests run before you push your changes. & you will not need to make a new pull request - once you have pushed this pr will be updated. Just drag into "for review"

File: client/subcomponents/volunteers/VolunteerButton/volunteerButtonHelper.js (hunk 1 above)
Comment (Frances-Hughes) [reply client/subcomponents/volunteers/VolunteerButton/volunteerButtonHelper.js:21]: Thanks! Have fixed it now, let us know if its all good :)

File: client/subcomponents/volunteers/VolunteerListItem/volunteerListItemHelper.js (hunk 4 above)
Comment (Frances-Hughes) [reply client/subcomponents/volunteers/VolunteerListItem/volunteerListItemHelper.js:10]: Thanks! Have fixed it now, let us know if its all good :)

File: client/views/admin/Events/AddEvent/addEventHelper.js (hunk 5 above)
Comment (Frances-Hughes) [reply client/views/admin/Events/AddEvent/addEventHelper.js:14]: Done!

**Review by johnny-sainsbury — APPROVED** [review]
Nice work :)

**Pull request merged**